src/test/kotlin/DirectoryLogicTester.kt

```diff
@@ -0,0 +1,52 @@
+import com.github.doyaaaaaken.kotlincsv.dsl.csvReader
+import org.junit.Test
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.assertThrows
+import java.io.File
+import java.lang.IllegalArgumentException
+
+class DirectoryLogicTester {
+
+    private val directoryLogic = DirectoryLogic()
+
+    @Test
+    fun testBuildFullFilePath() {
```

Comment:
This is a long function with mostly copy pasted code; can we try to restructure this to reuse that code?